src/functions/cancellations-data-endpoint/index.ts

```diff
@@ -0,0 +1,175 @@
+import {
+    createHttpNotFoundErrorResponse,
+    createHttpServerErrorResponse,
+    createHttpSuccessResponse,
+    createHttpUnauthorizedErrorResponse,
+    createHttpValidationErrorResponse,
+} from "@bods-integrated-data/shared/api";
+import { getCancellationsSubscription } from "@bods-integrated-data/shared/cancellations/utils";
+import { getDate } from "@bods-integrated-data/shared/dates";
+import { putDynamoItem } from "@bods-integrated-data/shared/dynamo";
+import { logger, withLambdaRequestTracker } from "@bods-integrated-data/shared/logger";
+import { putS3Object } from "@bods-integrated-data/shared/s3";
+import {
+    AvlSubscription,
+    HeartbeatNotification,
+    heartbeatNotificationSchema,
+} from "@bods-integrated-data/shared/schema";
+import { CancellationsSubscription } from "@bods-integrated-data/shared/schema/cancellations-subscribe.schema";
+import { SubscriptionIdNotFoundError } from "@bods-integrated-data/shared/utils";
+import { InvalidApiKeyError, createStringLengthValidation } from "@bods-integrated-data/shared/validation";
+import { APIGatewayProxyEvent, APIGatewayProxyHandler, Context } from "aws-lambda";
+import { XMLParser } from "fast-xml-parser";
+import { ZodError, z } from "zod";
+
+const requestParamsSchema = z.preprocess(
+    Object,
+    z.object({
+        subscriptionId: createStringLengthValidation("subscriptionId"),
+    }),
+);
+
+const requestBodySchema = z.string({
+    required_error: "Body is required",
+    invalid_type_error: "Body must be a string",
+});
+
+const arrayProperties = [
```

Comment:
There is a similar variable used in `src/functions/bods-disruptions-processor/index.ts` so it's worth creating a single, shared one in the shared folder that they both reference. The only property the other one has that is missing from this one is `AffectedNetwork`.